src/thunderbots/software/ai/primitive/stop_primitive.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "ai/primitive/primitive.h"
+
+class StopPrimitive : public Primitive
+{
+    public:
+     static const std::string PRIMITIVE_NAME;
+
+    /**
+     * Creates a new Stop Primitive
+     *
+     * @param robot_id The id of the Robot to run this Primitive
+     * @param stop to stop the robot or not
+     */
+     explicit StopPrimitive(unsigned int robot_id, bool stop);
+
+    /**
+     * Creates a new Stop Primitive from a Primitive message
+     *
+     * @param primitive_msg The message from which to create the Stop Primitive
+     */
+     explicit StopPrimitive(const thunderbots_msgs::Primitive &primitive_msg);
+
+     std::string getPrimitiveName() const override;
+
+     unsigned int getRobotId() const override;
+
+     std::vector<double> getParameterArray() const override;
+
+     std::vector<bool> getExtraBitArray() const override;
+
+    private:
+     unsigned int robot_id;
+     bool stop;
```

Comment:
I _think_ this boolean parameter is to enable coasting rather than enable stopping. I'm not somewhere where I can check rn, but that's also what we have in our primitive specs from #86. We should double check that and change it if necessary.

@henrymliu fyi

Reply:
Yes that's correct, one value coasts, the other applies reverse voltage to the motors
Can't remember which one is which